### PR TITLE
#2: Add Fastly as new provider

### PR DESCRIPTION
**Pull request by ssuda**
#2

### CODE REVIEW

File: test/controllers/FastlyCacheControllerTest.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        fastlyFlushMock = fastlyAPIMock
+          .intercept('/*', 'PURGE');
+        done();
+      });
+
+      it('should flush a complete zone of the Fastly cache', (done: Function) => {
+        fastlyFlushMock
+          .matchHeader('accept', 'application/json')
+          .matchHeader('authorization', 'U2FsdGVkX18D8TD+GD3REqc8cdjRikR6socyNOVSrN0=')
+          .reply(200, { success: true });
```
Comment (dax) [inline test/controllers/FastlyCacheControllerTest.ts]: Sorry but your tests are not working as this is a call signature to Fasterize API, not Fastly API.

**Comment by ssuda** [comment]
Hi David, fixed test case and changed flushZone to flushService

**Comment by dax** [comment]
Hi @ssuda, did you have some time to take a look at multi-CDN flush?

**Comment by ssuda** [comment]
@dax added fastly to multicachecontroller including test

File: test/controllers/MultiCacheControllerTest.ts [hunk 2]
```diff
@@ -35,6 +38,8 @@ describe('/v1/caches', () => {
           .get('/zones/purge/1.json');
         fasterizeFlushMock = fasterizeAPIMock
           .delete('/v1/configs/42/cache');
+        fastlyFlushMock = fastlyAPIMock
+          .post('/service/abcd/purge_all');
```
Comment (dax) [inline test/controllers/MultiCacheControllerTest.ts]: The mock for Fastly API is setup but no test use it thus there is no test that assert the code added here https://github.com/fasterize/darkstar/pull/4/commits/147bf6837af06224d8bde273a688e5ef9afa6d52#diff-0311da4455cc94ee09a0b49aac70e67eR62 is functional

Comment (ssuda) [reply test/controllers/MultiCacheControllerTest.ts]: @dax fixed the test case

**Pull request closed without merging**